File: Old Java Code/org/hardgate/familytree/formatters/MarriageOutputFormatter.java
/*
 * Created on 04-Jan-2005
 *
 */
package org.hardgate.familytree.formatters;

import java.io.PrintWriter;

import org.hardgate.familytree.registrations.MarriageRegistration;
import org.hardgate.familytree.registrations.Registration;

/**
 * @author db
 *
 */
public class MarriageOutputFormatter extends BaseOutputFormatter {

    /* (non-Javadoc)
     * @see org.hardgate.familytree.formatters.BaseOutputFormatter#printHeader(java.io.PrintWriter)
     */
    public void printHeader (PrintWriter output) {
        output.print("HusbandSurname,HusbandForenames,HusbandAge,HusbandOccupation,");
        output.print("HusbandResidence,HusbandBirthDate,HusbandBirthLocation,HusbandBestLocation,");
        output.print("HusbandsFathersName,HusbandsFathersOccupation,HusbandsFatherDeceased,");
        output.print("HusbandsMothersName,HusbandsMothersOccupation,HusbandsMotherDeceased,");
        output.print("WifeSurname,WifeForenames,WifeAge,WifeOccupation,");
        output.print("WifeResidence,WifeBirthDate,WifeBirthLocation,WifeBestLocation,");
        output.print("WifesFathersName,WifesFathersOccupation,WifesFatherDeceased,");
        output.print("WifesMothersName,WifesMothersOccupation,WifesMotherDeceased,");
        output.print("Religion,DateOfMarriage,PlaceOfMarriage");
        output.println();
     }

    /* (non-Javadoc)
     * @see org.hardgate.familytree.formatters.BaseOutputFormatter#printItem(org.hardgate.familytree.Registration, java.io.PrintWriter)
     */
    public void printItem (Registration reg, PrintWriter output) {
        MarriageRegistration m = (MarriageRegistration) reg;
        if (m.getGender().equals("M")) {
            printFamilyGroup(m.getRegistrationDate(), m.getFamilyGroup(), output);
            printFamilyGroup(m.getRegistrationDate(), m.getSpousesFamilyGroup(), output);
        } else {
            printFamilyGroup(m.getRegistrationDate(), m.getSpousesFamilyGroup(), output);
            printFamilyGroup(m.getRegistrationDate(), m.getFamilyGroup(), output);
        }
        output.print("\""); output.print(m.getReligion()); output.print("\",");
        output.print("\""); output.print(m.getDateOfMarriage()); output.print("\",");
        output.print("\""); output.print(m.getPlaceOfMarriage()); output.println("\"");
    }
}

File: FTAnalyser/Formatters/MarriageOutputFormatter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml;
using System.IO;

namespace FTAnalyser
{
    public class MarriageOutputFormatter : BaseOutputFormatter {

        public override void printHeader(StreamWriter output)
        {
            output.Write("HusbandSurname,HusbandForenames,HusbandAge,HusbandOccupation,");
            output.Write("HusbandResidence,HusbandBirthDate,HusbandBirthLocation,HusbandBestLocation,");
            output.Write("HusbandsFathersName,HusbandsFathersOccupation,HusbandsFatherDeceased,");
            output.Write("HusbandsMothersName,HusbandsMothersOccupation,HusbandsMotherDeceased,");
            output.Write("WifeSurname,WifeForenames,WifeAge,WifeOccupation,");
            output.Write("WifeResidence,WifeBirthDate,WifeBirthLocation,WifeBestLocation,");
            output.Write("WifesFathersName,WifesFathersOccupation,WifesFatherDeceased,");
            output.Write("WifesMothersName,WifesMothersOccupation,WifesMotherDeceased,");
            output.Write("Religion,DateOfMarriage,PlaceOfMarriage");
            output.WriteLine();
         }

        public override void printItem(Registration reg, StreamWriter output)
        {
            MarriageRegistration m = (MarriageRegistration) reg;
            if (m.Gender == "M") {
                printFamilyGroup(m.RegistrationDate, m.FamilyGroup, output);
                printFamilyGroup(m.RegistrationDate, m.SpousesFamilyGroup, output);
            } else {
                printFamilyGroup(m.RegistrationDate, m.SpousesFamilyGroup, output);
                printFamilyGroup(m.RegistrationDate, m.FamilyGroup, output);
            }
            output.Write("\""); output.Write(m.Religion); output.Write("\",");
            output.Write("\""); output.Write(m.DateOfMarriage); output.Write("\",");
            output.Write("\""); output.Write(m.PlaceOfMarriage); output.WriteLine("\"");
        }
    }
}